feat(CodeArts): add a new resource codearts deploy group (#3356)
huaweicloud/terraform-provider-huaweicloud: docs/resources/codearts_deploy_group.md

---

> subcategory: "CodeArts"

Reviewer: the subcategory should be **CodeArts Deploy**

Author: OK

Manages a CodeArts deploy group resource within HuaweiCloud.

## Example Usage

### Using proxy access mode

```hcl
variable "project_id" {}

resource "huaweicloud_codearts_deploy_group" "test" {
  project_id    = var.project_id
  name          = "test_group"
  os_type       = "linux"
  description   = "test description"
}
```

### Without using proxy access mode

```hcl
variable "project_id" {}

resource "huaweicloud_codearts_deploy_group" "test" {
  project_id    = var.project_id
  name          = "test_group"
  os_type       = "linux"
  description   = "test description"
  is_proxy_mode = 0
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `name` - (Required, String) Specifies the group name. The name consists of 3 to 128 characters, including letters,
  digits, chinese characters or `-_.` symbols.

* `project_id` - (Required, String, ForceNew) Specifies the project ID for CodeArts service.

  Changing this parameter will create a new resource.

> * `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid value are **windows** and **linux**.

Reviewer 2: ```suggestion
* `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid value is **windows** or **linux**.
```
or
```suggestion
* `os_type` - (Required, String, ForceNew) Specifies the operating system. Valid values are **windows** and **linux**.
```

Author: OK
